components/homme/src/share/prim_driver_base.F90

```diff
@@ -1356,16 +1376,22 @@ subroutine prim_step_flexible(hybrid, elem, nets, nete, dt, tl, hvcoord, compute
                 end do
              else
                 ! Set np1_qdp to -1 to remap dynamics variables but
-                ! not tracers.
-                call vertical_remap(hybrid, elem, hvcoord, dt_remap, tl%np1, -1, nets, nete)
+                ! not tracers
+                call vertical_remap(hybrid, elem, hvcoord, dt_remap, tl%np1, &
+		                    -1, nets_in, nete_in)
+
              end if
           end if
        end if
        ! defer final timelevel update until after Q update.
     enddo
     call t_stopf("prim_step_dyn")
 
-    if (qsize > 0) then
+    ! Update tracers based on LS vertical velocity
+    scm_update_T = .false.
+    call prim_step_scm(elem, nets_in, nete_in, dt, tl, hvcoord, scm_update_T)
```

Comment:
This line confuses me. It appears like prim_step_scm will always get here, i.e., even when single_column is false.

Reply:
Did the test `SMS_Ln5.ne4pg2_ne4pg2.FC5AV1C-L.cam-thetahy_sl_pg2` run when you ran `e3sm_developer`? It would alarm me if that test passes; it seems this unguarded call to prim_step_scm must be invalidating some data. Maybe the test doesn't run for enough time steps.